fix(KONFLUX-5886): allow unselecting default context

CryptoRodeo opened the pull request:
The user should be able to unselect the default context during the creation of an integration test.

Other changes:
- Contexts are now required (at least one)

CryptoRodeo commented:
Demo of the following:
- default context can now be unselected (main bug fixed)
- contexts are now required by the form and modal (at least 1 must be selected).

https://github.com/user-attachments/assets/6d713920-b569-4e43-b337-91c369705d64

CryptoRodeo commented:
It appears that the code coverage step fails:
```
error - 2024-12-18 21:48:17,362 -- Report creating failed: {"message":"Token required because branch is protected"}
Error: Codecov:
                        Failed to properly create report: The process '/home/runner/work/_actions/codecov/codecov-action/v4/dist/codecov' failed with exit code 1
```

testcara commented:
I met the same token problem for my ci jobs but after 2 days later, retry fixed the problem. It seems the ci is not very stable for this part.

testcara commented:
LGTM

JoaoPedroPP commented:
LGTM

StanislavJochman commented:
LGTM

sahil143 commented on src/components/IntegrationTests/utils/test-utils.ts:
```diff
@@ -0,0 +1,17 @@
+import { fireEvent, screen, act } from '@testing-library/react';
```
move this file under __tests__?

CryptoRodeo replied:
File has been moved

CryptoRodeo replied:
It looks like the test run failed because only tests are expected under `__tests__`... would moving this back to `utils`  be fine?

CryptoRodeo replied:
@sahil143 I reverted the change to fix the tests from failing (see above). If you have any recommendations on where else I should move this besides `__tests__` to prevent the failure again please let me know.

sahil143 replied:
My only concern is that the testing utilities should not be kept with the actual component utilities. I like the idea of creating separate utilities for opening dropdowns that can be used everywhere else. You can rename the functions to something more specific like `openIntegrationTestContextDropown` and move them to the common test-utils in src/utils/test-utils.ts. I am open to any other suggestions.

CryptoRodeo commented:
/retest

CryptoRodeo commented:
Code patch check looks safe to ignore IMO.

sahil143 commented on src/components/IntegrationTests/IntegrationTestForm/IntegrationTestView.tsx:
```diff
@@ -55,10 +56,20 @@ const IntegrationTestView: React.FunctionComponent<
-    if (!contexts?.length) return [];
+  const getFormContextValues = (
+    integrateTest: IntegrationTestScenarioKind | null | undefined,
+  ): FormContext[] => {
```
Adding unit tests for this utility should help with the codecoverage

CryptoRodeo replied:
Good catch! Test added.

CryptoRodeo commented:
@sahil143 That's fair. I moved the test utils to `src/utils/test-utils.ts` and renamed them: https://github.com/konflux-ci/konflux-ui/pull/58/files#diff-f13f3abfe65497faaec0c4ca81fa8037b6aa1094ed9e293becd345375fdcfda6R250-R261
